gcs_storage_client.go

```diff
@@ -329,7 +333,7 @@ func (util *snowflakeGcsClient) extractBucketNameAndPath(location string) *gcsLo
 	if strings.Contains(location, "/") {
 		containerName = location[:strings.Index(location, "/")]
 		path = location[strings.Index(location, "/")+1:]
-		if path != "" && strings.HasSuffix(location, "/") {
+		if path != "" && !strings.HasSuffix(path, "/") {
```

Comment:
This check was adding double "/" for download and not adding it for upload so both had the wrong URL.